### ISSUE TITLE
Type checking behaviour for RBG PRNG different when upgrading to jax v0.4.16

### ISSUE THREAD

**Issue by ayaka14732** [google/jax#17699]
### Description

When upgrading to jax v0.4.16, the following `DeprecationWarning` is shown:

```
DeprecationWarning: jax.random.PRNGKeyArray is deprecated. Use jax.Array for annotations, and jax.dtypes.issubdtype(arr, jax.dtypes.prng_key) for runtime detection of typed prng keys.
```

This implies that the behaviour of `isinstance(key, jax.random.PRNGKeyArray)` and `jax.dtypes.issubdtype(key, jax.dtypes.prng_key)` should be the same. However, they are not the same when using RBG PRNG.

Code to reproduce:

```python
import os; os.environ['JAX_PLATFORMS'] = 'cpu'

import jax
import jax.random as rand

jax.config.update('jax_enable_custom_prng', True)
jax.config.update('jax_default_prng_impl', 'rbg')

key = rand.PRNGKey(3407)
print(isinstance(key, jax.random.PRNGKeyArray))
print(jax.dtypes.issubdtype(key, jax.dtypes.prng_key))
```

Output:

```
True
False
```

### What jax/jaxlib version are you using?

jax 0.4.16

### Which accelerator(s) are you using?

CPU

### Additional system info

Python 3.11.5, Ubuntu 20.04.6 LTS x86_64

**Comment by jakevdp** [comment]
I think you meant this:
```python
jax.dtypes.issubdtype(key.dtype, jax.dtypes.prng_key)
```
With that change, the code should work as expected.

**Comment by jakevdp** [comment]
Looks like the snippet in the warning is wrong – I'll correct that.

You can read more about this in https://jax.readthedocs.io/en/latest/jep/9263-typed-keys.html

**Comment by jakevdp** [comment]
Fixed in #17709 – thanks for the report!
